cpp/src/arrow/filesystem/filesystem_test.cc

```diff
@@ -632,6 +633,13 @@ TEST_F(TestMockFS, FileSystemFromUri) {
   ASSERT_OK_AND_ASSIGN(fs_, FileSystemFromUri("mock:///foo/bar?q=zzz", &path));
   ASSERT_EQ(path, "foo/bar");
   CheckDirs({});
+  ASSERT_OK_AND_ASSIGN(fs_, FileSystemFromUri("mock:/folder+name/bar?q=zzz", &path));
```

Comment:
Should `+` in the path raise or is that expected for white space? In any case we should add URIs with white space to the test as this was reported in the issue.

Reply:
Well, there _is_ a test with whitespace below :-)

Reply:
Oh 🤦‍♀️  sorry about that :D